common/app/conf/Filters.scala

```diff
@@ -67,6 +67,27 @@ object SurrogateKeyFilter extends Filter with ExecutionContexts {
   }
 }
 
+object AmpFilter extends Filter with ExecutionContexts with implicits.Requests {
+  override def apply(nextFilter: (RequestHeader) => Future[Result])(request: RequestHeader): Future[Result] = {
+    if (request.isAmp) {
+      val domain = request.headers.get("Origin") match {
+        case None => {
+          "https://" + request.domain
+        }
+        case Some(requestOrigin) => {
+          requestOrigin
+        }
+      }
+      val exposeAmpHeader = "Access-Control-Expose-Headers" -> "AMP-Access-Control-Allow-Source-Origin"
+      val ampHeader = "AMP-Access-Control-Allow-Source-Origin" -> Configuration.amp.corsOrigins.find(_ == domain).getOrElse("*")
```

Comment:
I know `*` is set as a fallback in case the domain cannot be found in our config. But given this is not recommended should we log/track if this happens. That could allow us to monitor it and add a domain in case Google makes any changes for instance